pkg/core/native/native_neo.go

```diff
@@ -1042,24 +1083,21 @@ func (n *NEO) getAccountState(ic *interop.Context, args []stackitem.Item) stacki
 	return item
 }
 
-// ComputeNextBlockValidators returns an actual list of current validators.
-func (n *NEO) ComputeNextBlockValidators(blockHeight uint32, d *dao.Simple) (keys.PublicKeys, error) {
-	numOfCNs := n.cfg.GetNumOfCNs(blockHeight + 1)
-	// Most of the time it should be OK with RO cache, thus try to retrieve
-	// validators without RW cache creation to avoid cached values copying.
+// ComputeNextBlockValidators computes an actual list of current validators that is
+// relevant for the latest persisted block and based on the latest changes made by
+// register/unregister/vote calls.
+// Note: this method isn't actually "computes" new committee list and calculates
+// new validators list from it. Instead, it uses cache, but the cache itself is
+// updated every block.
+func (n *NEO) ComputeNextBlockValidators(d *dao.Simple) keys.PublicKeys {
```

Comment:
I've reverted the `ComputeNextBlockValidators` -> `NextBlockValidators` renaming, because the name of this exported method must match its meaning. And what this method performs is literally computing validators from based on the most fresh vote/candidates ratio got from the latest contract storage. The users of this exported method don't care *how* it is done, they just have to now that these data are fresh. And it's just NAO contract who stores this value in cache and updates it every PostPersist.

Reply:
Call it `GetNextNextBlockValidators()`. Make `validators` always have the list (maybe call them `nextNextValidators, never set it to `nil`.